File: Btech.Sql.Console/Models/Responses/Base/PaginationResponse.cs
using Newtonsoft.Json;

namespace Btech.Sql.Console.Models.Responses.Base;

public class PaginationResponse<T> : Response
    where T : class
{
    #region Public Constants

    public const string PageJsonPropertyName = "currentPage";
    public const string PerPageJsonPropertyName = "perPage";
    public const string TotalAmountJsonPropertyName = "totalAmount";
    public const string EntitiesJsonPropertyName = "entities";

    #endregion Public Constants

    #region Public Properties

    [JsonProperty(PageJsonPropertyName, NullValueHandling = NullValueHandling.Ignore)]
    public int? CurrentPage { get; set; }

    [JsonProperty(PerPageJsonPropertyName, NullValueHandling = NullValueHandling.Ignore)]
    public int? PerPage { get; set; }

    [JsonProperty(TotalAmountJsonPropertyName, NullValueHandling = NullValueHandling.Ignore)]
    public long? TotalAmount { get; set; }

    [JsonProperty(EntitiesJsonPropertyName, NullValueHandling = NullValueHandling.Ignore)]
    public List<T> Entities { get; set; }

    #endregion Public Properties
}

File: Btech.Sql.Console/ClientApp/src/app/_models/responses/base/pagination-response.ts
import {ResponseBase} from './response';

export class PaginationResponse<T> extends ResponseBase {
    private _currentPage?: number;
    private _perPage?: number;
    private _totalAmount?: number;
    private _entities?: T[];

    get currentPage(): number | undefined {
        return this._currentPage;
    }

    get perPage(): number | undefined {
        return this._perPage;
    }

    get totalAmount(): number | undefined {
        return this._totalAmount;
    }

    get entities(): T[] | undefined {
        return this._entities;
    }

    set currentPage(value: number | undefined) {
        this._currentPage = value;
    }

    set perPage(value: number | undefined) {
        this._perPage = value;
    }

    set totalAmount(value: number | undefined) {
        this._totalAmount = value;
    }

    set entities(value: T[] | undefined) {
        this._entities = value;
    }

}